### ISSUE TITLE
requests are not always logged in IIS

### ISSUE THREAD

**Issue by ofersagi** [aspnet/IISIntegration#339]
i managed to reproduce a situation when a request to iis that was routed through aspnetcore module to aspnetcore app was not written to IIS log file.

this happend when an exception was thrown inside a background task that the main code was waited to complete.

in order to reproduce:

        Task<IActionResult>[] taskList = new Task<IActionResult>[numberOfThreadTasksRequired];

loop twice or more with 
```
           Task.Run(() =>
            {
              // make something wrong... like writing same header more than once
                         Response.Headers.Add("x-something", "foo");
// this causing exception and the request is not logged in IIS :(

            });

        Task.WaitAll(taskList);
```

**Comment by Tratcher** [comment]
I think you're missing a few lines of code. Can you fill it in so we're sure what you're doing?

**Comment by ofersagi** [comment]
the basic idea is demonstrated. just make some list of tasks and wait on them. then do the bad thing inside.

**Comment by ofersagi** [comment]
reproduced it again today. can send a fiddler archive + iis + aspnetcore log files privately.
actually the subject should be IIS logs are not being written consistently. sometimes requests are just not logged even after IISRESET new reqeusts are being logged while some requests wasn't logged.

**Comment by davidfowl** [comment]
`Task<IActionResult>[] taskList = new Task<IActionResult>[numberOfThreadTasksRequired];`

What sorta code did you write that ended up with a list of `IActionResult`? Are you implementing custom child actions or something?

**Comment by ofersagi** [comment]
it's sorta bulk operation over some other api/action.

**Comment by muratg** [comment]
@ofersagi could you provide a simple repro app and post it on GitHub?
